db/compaction/compaction_picker_fifo.cc

```diff
@@ -71,10 +71,13 @@ Compaction* FIFOCompactionPicker::PickTTLCompaction(
   if (current_time > mutable_cf_options.ttl) {
     for (auto ritr = level_files.rbegin(); ritr != level_files.rend(); ++ritr) {
       FileMetaData* f = *ritr;
-      uint64_t creation_time = f->TryGetFileCreationTime();
-      if (creation_time == kUnknownFileCreationTime ||
-          creation_time >= (current_time - mutable_cf_options.ttl)) {
-        break;
+      if (f->fd.table_reader && f->fd.table_reader->GetTableProperties()) {
+        uint64_t creation_time =
+            f->fd.table_reader->GetTableProperties()->creation_time;
```

Comment:
I guess `f->fd.table_reader->GetTableProperties()` is never null? We can assert it.

Reply:
This requires max_open_files to be -1.

Reply:
You are right. We should assert it.

Reply:
Let's keep the `if` here. In the future, we want to handle the case in which f->fd.table_reader is null.